### ISSUE TITLE
Oni freezes sometimes whenever switching/opening files

### ISSUE THREAD

**Issue by cyansprite** [onivim/oni#664]
We get this exception...
```
internal/process/next_tick.js:103 Uncaught (in promise) Syntax highlighting was for different file.
```
Then it freezes oni, which the only recovery seems to close it.

**Comment by extr0py** [comment]
Interesting - we could try disabling the syntax highlighting (that Oni adds), and see if that helps-

A few other questions:
- Which platform is this on? Windows?
- What kind of file is it? Is it large?
- Is it a different file everytime? Or does it happen consistently for certain files?
- How often does it occur?
- Is it something you just started seeing recently, or has it been happening since you started using Oni?

Hopefully this can help narrow it down.

I have seen some freezing issues when I loaded a file that had a giant base64 data string in it. I wonder if it is related to that?

Another issue that could be happening, is if the file is large, it might be spending a lot of time sending us the `buffer-update` event, or Oni might be choking on receiving it.

There also cases where neovim has been crashing on Windows: #409 - this could also be related.

I need to have a better way to disable the Oni-side plugins as well as the buffer-updates so that we can better rule these out (or identify them as problematic).

**Comment by cyansprite** [comment]
>Which platform is this on? Windows?

Yes

> What kind of file is it? Is it large?

Just a typescript file in our repo, probably between 40~400 lines.

> Is it a different file everytime? Or does it happen consistently for certain files?

As far as I can tell it's random, it's just whenever I'm testing my changes or coding in our repo.

> How often does it occur?

Intermittently, around 1/3 times I use Oni.

> Is it something you just started seeing recently, or has it been happening since you started using Oni?

I would say it's something I've always seen, before it would crash I think? Now it hangs.

> Another issue that could be happening, is if the file is large, it might be spending a lot of time sending us the buffer-update event, or Oni might be choking on receiving it.

Hmm... I'm not sure, I only think it has ever happened with TypeScript files.

> I need to have a better way to disable the Oni-side plugins as well as the buffer-updates so that we can better rule these out (or identify them as problematic).

I think perhaps this may help, I think it may be the language server protocol, but that's just conjecture.

**Comment by extr0py** [comment]
Thanks for the responses!

One option might be to disable the typescript integration, just to see if that is the culprit.... A quick and dirty way to do that is to delete the `oni-plugin-typescript` folder in vim\core, and see if that makes the problem go away (unfortunately, you lose completion and language functionality there)...

If it turns out that is the culprit, we could bring it back, but disable some pieces of it.

My hunch is that these issues are either due to the way we send buffer updates from neovim -> Oni, or to the syntax highlighting we apply from the language server side.

**Comment by bryphe** [comment]
This should be fixed now with Neovim 0.2.1 (and 0.2.2) - I haven't seen it happen again since the upgrade.
